SingleSource/UnitTests/Atomic/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+set(FP_TOLERANCE 1000000)
+execute_process(COMMAND ${CMAKE_C_COMPILER} --print-file-name=libclang_rt.atomic.so
+                OUTPUT_VARIABLE _path_to_libatomic
+                OUTPUT_STRIP_TRAILING_WHITESPACE)
+get_filename_component(_libatomic_dir ${_path_to_libatomic} DIRECTORY)
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${_path_to_libatomic} -Wl,-rpath=${_libatomic_dir}")
```

Comment:
Changing global variables is usually not a good idea. Prefer [`target_link_options`](https://cmake.org/cmake/help/latest/command/target_link_options.html), or at the very least or at least [add_link_options](https://cmake.org/cmake/help/latest/command/add_link_options.html) so it applies to all targets in this CMakeLists and subdirectories.

Please add a comment on why it is needed.